felix/dataplane/linux/endpoint_mgr.go

```diff
@@ -669,6 +669,14 @@ func (m *endpointManager) resolveWorkloadEndpoints() {
 	}
 
 	removeActiveWorkload := func(logCxt *log.Entry, oldWorkload *proto.WorkloadEndpoint, id types.WorkloadEndpointID) {
+		// QoS state must be removed before the workload itself is removed
+		if oldWorkload != nil {
+			logCxt.Info("deleting QoS bandwidth state if present")
+			err := m.maybeUpdateQoSBandwidth(oldWorkload, nil)
+			if err != nil {
+				logCxt.WithField("workload", oldWorkload).Infof("error deleting QoS bandwidth state: %v", err)
```

Comment:
Given the "must" in the comment above, what is the position if there is an error here?

Reply:
Fair point, I guess the comment should not say "must". An error here usually means that the CNI plugin removed the interface before felix would, and the QoS config goes away in a perfectly fine manner with it...